File: examples/ruby/fetch_offers.rb
require_relative 'solid_jobs_client'
require_relative 'helpers'

client = SolidJobsClient.new
campaign = 'ruby-client'

# 1) Basic listing — first page of IT offers
first_page = client.get_offers('IT', campaign, page_size: 5)
puts "Found #{first_page['totalCount']} offers (pages: #{first_page['totalPages']})."

first_page['jobs'].each do |job|
  puts "#{job['title']} @ #{job['company']} [#{job['experienceLevel']}] (#{format_salary(job['salary'])})"
end

# 2) Filtering + sorting — Senior Java in Warsaw, highest salary first
senior_java = client.get_offers('IT', campaign,
  search_terms: ['java'],
  cities: ['Warszawa'],
  sub_categories: ['Java'],
  experiences: ['Senior'],
  minimum_salary: 15_000,
  sort_active: 'salaryTo',
  sort_direction: 'desc',
  page_size: 10
)

puts "\nSenior Java in Warsaw: #{senior_java['totalCount']} offers"
senior_java['jobs'].each do |job|
  puts "- #{job['title']} (#{format_salary(job['salary'])})"
end

# 3) Iterate all .NET offers
puts "\nAll .NET offers:"
all_dotnet = client.get_all_offers('IT', campaign, sub_categories: ['DotNet'])
all_dotnet.each_with_index do |job, i|
  printf "%3d. %s @ %s\n", i + 1, job['title'], job['company']
end
puts "Total: #{all_dotnet.size}"

File: examples/php/fetch_offers.php
<?php

require_once __DIR__ . '/SolidJobsClient.php';
require_once __DIR__ . '/helpers.php';

$client = new SolidJobsClient();
$campaign = 'php-client';

// 1) Basic listing — first page of IT offers
$firstPage = $client->getOffers('IT', $campaign, ['pageSize' => 5]);
echo "Found {$firstPage['totalCount']} offers (pages: {$firstPage['totalPages']}).\n";

foreach ($firstPage['jobs'] as $job) {
    $sal = formatSalary($job['salary']);
    echo "{$job['title']} @ {$job['company']} [{$job['experienceLevel']}] ($sal)\n";
}

// 2) Filtering + sorting — Senior Java in Warsaw, highest salary first
$seniorJava = $client->getOffers('IT', $campaign, [
    'searchTerms' => ['java'],
    'cities' => ['Warszawa'],
    'subCategories' => ['Java'],
    'experiences' => ['Senior'],
    'minimumSalary' => 15000,
    'sortActive' => 'salaryTo',
    'sortDirection' => 'desc',
    'pageSize' => 10,
]);

echo "\nSenior Java in Warsaw: {$seniorJava['totalCount']} offers\n";
foreach ($seniorJava['jobs'] as $job) {
    echo "- {$job['title']} (" . formatSalary($job['salary']) . ")\n";
}

// 3) Iterate all .NET offers
echo "\nAll .NET offers:\n";
$allDotNet = $client->getAllOffers('IT', $campaign, ['subCategories' => ['DotNet']]);
foreach ($allDotNet as $i => $job) {
    printf("%3d. %s @ %s\n", $i + 1, $job['title'], $job['company']);
}
echo "Total: " . count($allDotNet) . "\n";
